src/scripts/main.js

```diff
@@ -1,3 +1,226 @@
 'use strict';
 
-// write your code here
+const startButton = document.querySelector('.start');
+const gameScore = document.querySelector('.game-score');
+const gameStatus = document.querySelector('.message-container');
+const table = document.querySelector('.game-field');
+const rows = table.rows;
+let score = 0;
+
+const gameField = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+function resetTheGame() {
+  score = 0;
+
+  for (let row = 0; row < gameField.length; row++) {
+    for (let cell = 0; cell < gameField.length; cell++) {
+      gameField[row][cell] = 0;
+    }
+  }
+}
+
+function changingDivMessage(value) {
+  for (const child of gameStatus.children) {
+    if (child.classList.contains(`message-${value}`)) {
+      child.classList.remove('hidden');
+    } else {
+      child.classList.add('hidden');
+    }
+  }
+}
+
+function randomNumber() {
+  let randomNum = Math.floor(Math.random() * (10 - 1) + 1);
+  let randomRowIndex = Math.floor(Math.random() * (4));
+  let randomColumnIndex = Math.floor(Math.random() * (4));
+
+  while (gameField[randomRowIndex][randomColumnIndex] !== 0) {
+    randomRowIndex = Math.floor(Math.random() * (4));
+    randomColumnIndex = Math.floor(Math.random() * (4));
+  }
+
+  if (randomNum < 2) {
+    randomNum = 4;
+  } else {
+    randomNum = 2;
+  }
+
+  gameField[randomRowIndex][randomColumnIndex] = randomNum;
+}
+
+function renderGameField() {
+  gameScore.innerText = score;
+
+  for (let x = 0; x < gameField.length; x++) {
```

Comment:
Consider using Array.prototype.forEach() method for iterating over the gameField array instead of the for loop. It will make the code more functional and easier to read.

You can apply it everywhere